File: peek_plugin_graphdb/plugin-module/GraphDbSegmentLinkTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {graphDbTuplePrefix} from "./_private/PluginNames";


@addTupleType
export class GraphDbSegmentLinkTuple extends Tuple {
    public static readonly tupleName = graphDbTuplePrefix + "GraphDbSegmentLinkTuple";

    //  he key of the vertex that joins the two segments
    vertexKey: string;

    //  The segment that this segment links to
    segmentKey: string;

    constructor() {
        super(GraphDbSegmentLinkTuple.tupleName)
    }
}

File: peek_plugin_graphdb/tuples/GraphDbSegmentLinkTuple.py
from vortex.Tuple import Tuple, addTupleType, TupleField

from peek_plugin_graphdb._private.PluginNames import graphDbTuplePrefix
from peek_plugin_graphdb._private.storage.GraphDbModelSet import GraphDbModelSet


@addTupleType
class GraphDbSegmentLinkTuple(Tuple):
    """ Segment Link Tuple

    This tuple is the publicly exposed Segment Links

    """
    __tupleType__ = graphDbTuplePrefix + 'GraphDbSegmentLinkTuple'

    #:  The key of the vertex that joins the two segments
    vertexKey: str = TupleField()

    #:  The segment that this segment links to
    segmentKey: GraphDbModelSet = TupleField()
